lib/poller.js

```diff
@@ -289,8 +289,7 @@ class Poller {
               allowed, reason
             }
           }
-        }
-        if (!reNaming.test(secretProperty.key)) {
+        } else if (!reNaming.test(secretProperty.key)) {
```

Comment:
I'm thinking its better we check for the respective keys and validate them if present.

```js
      externalData.forEach((secretProperty, index) => {
        if ('path' in secretProperty && !reNaming.test(secretProperty.path)) {
          allowed = false
          reason = `path ${secretProperty.path} does not match naming convention ${namingConvention}`
          return {
            allowed, reason
          }
        }

        if ('key' in secretProperty && !reNaming.test(secretProperty.key)) {
          allowed = false
          reason = `key name ${secretProperty.key} does not match naming convention ${namingConvention}`
          return {
            allowed, reason
          }
        }
      })
```

They never should be present at the same time, but if validation or other layer of the code allows for it, then this would potentially allow bypassing the validation. By supplying a bogus path that would validate and using a key which is ignored by validation.

Reply:
Yeah, checking `path` and `key` independently makes sense. I updated the code to do that.